src/components/card.js

```diff
@@ -1,53 +1,76 @@
 export default class Card {
-  constructor(cardData, templateSelector, handleImageClick) {
+  constructor(
+    cardData,
+    templateSelector,
+    handleImageClick,
+    handleDeleteCard,
+    handleLikeButton
+  ) {
     this._cardData = cardData;
     this._templateSelector = templateSelector;
     this._link = cardData.link;
     this._name = cardData.name;
-    this._element = null;
+    this._cardId = cardData._id;
+    this._likes = cardData.likes;
     this._handleImageClick = handleImageClick;
+    this._handleLikeButton = handleLikeButton;
+    this._handleDeleteCard = handleDeleteCard;
+    this._element = null;
+    this._likeButton = null;
+    this._like = cardData.isLiked || false; // Initialize like status from API
   }
 
   _getTemplate() {
     const cardTemplate = document.querySelector(this._templateSelector);
-    const cardElement = cardTemplate.content.cloneNode(true);
-    return cardElement.querySelector(".card");
+    const cardElement = cardTemplate.content
+      .cloneNode(true)
+      .querySelector(".card");
+    return cardElement;
+  }
+
+  _handleLikeIcon() {
+    if (this._like) {
+      this._likeButton.classList.add("card__like-button_active");
+    } else {
+      this._likeButton.classList.remove("card__like-button_active");
+    }
+  }
+
+  handleLike(liked) {
+    console.log("Updating like status:", liked);
+    this._like = liked;
+    this._handleLikeIcon();
   }
 
   _setEventListeners() {
-    const likeButton = this._element.querySelector(".card__like-button");
-    likeButton.addEventListener("click", () => {
-      likeButton.classList.toggle("card__like-button_active");
+    this._likeButton = this._element.querySelector(".card__like-button");
+    this._likeButton.addEventListener("click", () => {
+      console.log(`Like button clicked for cardId: ${this._cardId}`);
+      this._handleLikeButton(this._cardId, this);
     });
 
-    const deleteButton = this._element.querySelector(".card__delete-button");
-    deleteButton.addEventListener("click", () => {
-      this._handleDeleteCard();
-    });
+    this._element
+      .querySelector(".card__delete-button")
+      .addEventListener("click", () => {
+        this._handleDeleteCard(this._cardId, this._element);
+      });
 
     const cardImageEl = this._element.querySelector(".card__image");
```

Comment:
**Could be improved**

All elements found **every time** you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `getView`) only **1 time** in order not to waste resources on searching them again and again.

`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form`, `this._likeButton`, and so on.

```js
  getView() {
    this._element = this._getTemplate();
    this._likeButton = this._element.querySelector(".card__like-button");
    this._cardImage = this._element.querySelector(".card__image");
    this._trashButton = this._element.querySelector(".card__trash-button");

```